### DISCUSSION TITLE
Polymorphic Associations appear not to be created.

### DISCUSSION THREAD

**Discussion by Tzvetelin88** [sequelize/sequelize#14625] [Help & Questions]
Hello, I'm following documentation here: https://sequelize.org/docs/v6/advanced-association-concepts/polymorphic-associations/, 
to create two simple models with associations/relations between them, to get an access to Polymorphic/Helper association methods. But it seems something is not right. Helper method "createComment" -> _Property 'createComment' does not exist on type 'Image'_
Example:
```import { Sequelize, DataTypes, Model } from "sequelize";

export const sequelize = new Sequelize({
  dialect: "postgres",
  host: "localhost",
  port: 5432,
  database: "demo-test",
  username: "demo",
  password: "demo",
});

class Image extends Model {}
Image.init(
  {
    title: DataTypes.STRING,
    url: DataTypes.STRING,
  },
  { sequelize, modelName: "image" }
);

class Comment extends Model {}
Comment.init(
  {
    title: DataTypes.STRING,
    commentableId: DataTypes.INTEGER,
    commentableType: DataTypes.STRING,
  },
  { sequelize, modelName: "comment" }
);

Image.hasMany(Comment, {
  foreignKey: "commentableId",
  constraints: false,
  scope: {
    commentableType: "image",
  },
});
Comment.belongsTo(Image, {
  foreignKey: "commentableId",
  constraints: false,
});

const image = await Image.create({ url: "https://placekitten.com/408/287" });

const comment = await image.createComment({ content: "Awesome!" });
```

**Comment by Tzvetelin88** [comment]
I found the issue to be Typescript related, and Helper/Polymorphic do not appear in IDE.
If we put breakpoint and call Model "Image" into the Debug Console, all Helper/Polymorphic are there.

So, the solution was to add "any" as type when we get the model:
const image: any = await Image.create({ url: "https://placekitten.com/408/287" });
and then to use:
await image.createComment etc.

**Comment by ephys** [comment]
Please see https://sequelize.org/docs/v6/other-topics/typescript/ to view how to declare your models when you use typescript
